src/http/request.ts

```diff
@@ -27,8 +27,39 @@ const printWarnings = {
 }
 
 /** @internal */
-export function defineHttpRequest(envMiddleware: Middlewares): HttpRequest {
+export function defineHttpRequest(
+  envMiddleware: Middlewares,
+  {maxRetries, retryDelay}: {maxRetries: number; retryDelay?: (attemptNumber: number) => number}
+): HttpRequest {
   const request = getIt([
+    retry({
+      retryDelay: retryDelay as () => number, // This option is typed incorrectly in get-it.
+      maxRetries,
+      shouldRetry(err, attempt, options) {
```

Comment:
Could we export this for people that want to reuse it in their custom `shouldRetry` functions?